docs/source/methods/random_numbers.rst

```diff
@@ -62,11 +66,126 @@ Note that equation :eq:`lcg-skipahead` has the same general form as equation :eq
+
+A permuted congruential generator (PCG) aims to improve statistical quality 
+of a LCG by using permutation functions. The general algorithm consists of 
+two steps:
+
+1. advance the LCG generator according to :eq:`pcg_lcg`,
+2. output the LCG state "scrambled" by permutation function.
+
+
+
+OpenMC uses the PCG-RXS-M-XS variant with 64-bit state and 
+64-bit output. The code for permutation functiom is taken 
+from `PCG GitHub`_. The exact algorithm follows.
+
+1. Generate LCG output :math:`\xi_{i+1}`
+
+  .. math::
+      :label: pcg_lcg
+  
+      \xi_{i+1} = g \xi_i + c
+
+  where :math:`g=6364136223846793005` and :math:`c=1442695040888963407`.
+
+2. Apply random xorshift. 
+    * First, take the upper 5 bits of :math:`\xi_{i+1}` 
+      using the `arithmetic right shift operator`_ by 59 positions.
+    * Second, add :math:`5` and shift seed by that number of bits. The 
+      name "random" refers to this step, where the length of the shift 
+      is evaluated from the seed, not given as constant.
+    * Third, perform `bitwise XOR`_ with the upper bits and original :math:`\xi_{i+1}`.
+
+#. Multiply by :math:`6364136223846793005`.
+#. Apply xorshift - similarly to step 2, use `arithmetic right shift operator`_ 
+   to take upper 21 bits and perform `bitwise XOR`_. The length of 
+   the shift is constant here.
+#. Convert to double from interval :math:`[0, 1)`.
+
+
+**Advantages of PCG over LCG include**
+
+* increased statistical quality - measured by statistical tests from BigCrush library,
+* small performance burden compared to LCG.
+
+For elaborated description, see [ONeill]_.
+
+
+Example of PCG algorithm
+------------------------
+
+As this might be difficult to imagine, let's add an example. 
+
+* Let's assume seed value :math:`\xi_{i} = 1`.
+* From point (1), :math:`\xi_{i+1} = 7806831264735756412` according to :eq:`pcg_lcg`, which 
+  is::
+    
+    0110 1100 0101 0111 0110 1111 1010 1100 0100 0011 1111 1101 0000 0000 0111 1100
+
+  in binary representation.
+* The random xorshift from point (2) is performed in 3 steps
+   * First, bit shift by 59 positions means we keep first 5 bits of :math:`\xi_{i+1}`, which is
+     ``0 1101`` or :math:`13`, when represented as integer.
+   * Second, adding 5, we have to shift :math:`\xi_{i+1}` by :math:`13+5=18` bits - or, equivalently, 
+     throw away the last 18 bits, which yields::
+
+        01 1011 0001 0101 1101 1011 1110 1011 0001 0000 1111 1111
+        
+     which is :math:`29780697878783` as integer. 
+   * Finally, perform `bitwise XOR`_ with shifted bits and original :math:`\xi_{i+1}`::
+        
+        0110 1100 0101 0111 0110 1111 1010 1100 0100 0011 1111 1101 0000 0000 0111 1100
+        0000 0000 0000 0000 0001 1011 0001 0101 1101 1011 1110 1011 0001 0000 1111 1111
+        -------------------------------------------------------------------------------
+        0110 1100 0101 0111 0111 0100 1011 1001 1001 1000 0001 0110 0001 0000 1000 0011 
```

Comment:
This looks good in HTML, but overflows line in LaTeX pdf (which is ugly). 
If anyone knows how to make it a bit smaller, any suggestions welcome.

Reply:
Resolved by using `.. code-block:: text` outside of bullet list